### ISSUE TITLE
Blockly.bumpIntoBoundsHandler_ has doubled loading times

### ISSUE THREAD

**Issue by NeilFraser** [google/blockly#4871]
**Describe the bug**

When a large number of top-level blocks are loaded, `Blockly.bumpIntoBoundsHandler_` is called on each block in turn.  This consumes as much or more CPU time as loading/rendering the block itself.

**To Reproduce**

1. Go to the playground.
2. Open the console.
3. Execute: airstrike(2000)

**Expected behaviour**

The 2000 blocks should appear after about 5 seconds.  They should be ready to interact with at this time.

** Actual behaviour **

The 2000 blocks appear after about 5 seconds.  But the browser remains frozen for an additional five seconds while `bumpIntoBoundsHandler_` is executed 2000 times.  No interaction is possible until these calls complete.

**Desktop (please complete the following information):**
 - OS: MacOS
 - Browser Chrome and Firefox

**Comment by NeilFraser** [comment]
FYI, here's the stack trace for each call:
* Blockly.bumpIntoBoundsHandler_
* Blockly.MetricsManager.prototype.getScrollMetrics
* Blockly.MetricsManager.prototype.getContentMetrics
* Blockly.WorkspaceSvg.prototype.getBlocksBoundingBox
* Blockly.BlockSvg.prototype.getBoundingRectangle
* Blockly.BlockSvg.prototype.getRelativeToSurfaceXY
* Blockly.utils.getRelativeXY

It's `Blockly.utils.getRelativeXY` that appears to be consuming all the CPU.

**Comment by alschmiedt** [comment]
Is this a regression for both movable and immovable workspaces? 

#4872 should fix the problem for a movable workspace, but not an immovable workspace.

**Comment by NeilFraser** [comment]
Yes, at random I checked out the rc_2020_6 branch, and there was no UI delay after the blocks appear in either movable or immovable modes.

**Comment by alschmiedt** [comment]
Marking this as low priority since #4872 fixed a majority of the use cases. However, it is still a regression and I would like to understand what is causing the issue.
